examples/flutter_gallery/lib/demo/material/reorderable_list_demo.dart

```diff
@@ -0,0 +1,195 @@
+
+enum _ReorderableListType {
+  /// A list tile that contains a single line of text.
+  horizontalAvatar,
+
+  /// A list tile that contains a [CircleAvatar] followed by a single line of text.
+  verticalAvatar,
+
+  /// A list tile that contains three lines of text.
+  threeLine,
+}
+
+class ReorderableListDemo extends StatefulWidget {
+  const ReorderableListDemo({ Key key }) : super(key: key);
+
+  static const String routeName = '/material/reorderable-list';
+
+  @override
+  _ListDemoState createState() => new _ListDemoState();
+}
+
+class _ListDemoState extends State<ReorderableListDemo> {
+  static final GlobalKey<ScaffoldState> scaffoldKey = new GlobalKey<ScaffoldState>();
+
+  PersistentBottomSheetController<Null> _bottomSheet;
+  _ReorderableListType _itemType = _ReorderableListType.threeLine;
+  bool _reverseSort = false;
+  List<String> items = <String>[
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N',
+  ];
+
+  void changeItemType(_ReorderableListType type) {
+    setState(() {
+      _itemType = type;
+    });
+    _bottomSheet?.setState(() { });
+  }
+
+  void _showConfigurationSheet() {
+    final PersistentBottomSheetController<Null> bottomSheet = scaffoldKey.currentState.showBottomSheet((BuildContext bottomSheetContext) {
+      return new Container(
+        decoration: const BoxDecoration(
+          border: const Border(top: const BorderSide(color: Colors.black26)),
+        ),
+        child: new ListView(
+          shrinkWrap: true,
+          primary: false,
+          children: <Widget>[
+            new MergeSemantics(
+              child: new ListTile(
+                dense: true,
+                title: const Text('Horizontal Avatars'),
+                trailing: new Radio<_ReorderableListType>(
+                  value:_ReorderableListType.horizontalAvatar,
+                  groupValue: _itemType,
+                  onChanged: changeItemType,
+                )
+              ),
+            ),
+            new MergeSemantics(
+              child: new ListTile(
+                dense: true,
+                title: const Text('Vertical Avatars'),
+                trailing: new Radio<_ReorderableListType>(
+                  value: _ReorderableListType.verticalAvatar,
+                  groupValue: _itemType,
+                  onChanged: changeItemType,
+                )
+              ),
+            ),
+            new MergeSemantics(
+              child: new ListTile(
+                dense: true,
+                title: const Text('Three-line'),
+                trailing: new Radio<_ReorderableListType>(
+                  value: _ReorderableListType.threeLine,
+                  groupValue: _itemType,
+                  onChanged: changeItemType,
+                ),
+              ),
+            ),
+          ],
+        ),
+      );
+    });
+
+    setState(() {
+      _bottomSheet = bottomSheet;
+    });
+
+    _bottomSheet.closed.whenComplete(() {
+      if (mounted) {
```

Comment:
I just copied this directly from the other list demo.

This is done because the menu icon for showing the bottom sheet becomes disabled when the sheet is not null. Setting state to change it back to null is necessary to enable it again.